### ISSUE TITLE
debug ice: E0061 with several fn arguments where 0 are expected

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#108249]
### Code

This triggers a debug assertion due to bad spans when more than one input is given to a function that expects none.

```Rust
fn main() {
	Vec::new(0, 1);
}
```

without debug assertions, this results in 

````
error[E0061]: this function takes 0 arguments but 2 arguments were supplied
   --> extern-prelude.rs:2:2
    |
2   |     Vec::new(0, 1);
    |     ^^^^^^^^ -  - unexpected argument of type `{integer}`
    |              |
    |              unexpected argument of type `{integer}`
    |
note: associated function defined here
   --> /home/matthias/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:424:18
    |
424 |     pub const fn new() -> Self {
    |                  ^^^
help: remove the extra arguments
    |
2   -     Vec::new(0, 1);
2   +     Vec::new();
````

I think under the hood rustc might try to generate empty spans for within the `()` where it suggests to remove the fn args..? Just a wild guess though :thinking: 

### Meta

`rustc --version --verbose`:
```
eebdfb55fce148676c24555505aebf648123b2de
```

### Error output

```
error[E0061]: this function takes 0 arguments but 2 arguments were supplied
   --> extern-prelude.rs:2:2
    |
2   |     Vec::new(0, 1);
    |     ^^^^^^^^ -  - unexpected argument of type `{integer}`
    |              |
    |              unexpected argument of type `{integer}`
    |
note: associated function defined here
   --> /home/matthias/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:424:18
    |
424 |     pub const fn new() -> Self {
    |                  ^^^
help: remove the extra arguments
    |
2   -     Vec::new(0, 1);
2   +     Vec::new();
    |

error: aborting due to previous error

For more information about this error, try `rustc --explain E0061`
```

[ice.log](https://github.com/rust-lang/rust/files/10778191/ice.log)

**Comment by matthiaskrgr** [comment]
I wonder if this is a duplicate of https://github.com/rust-lang/rust/issues/108242 but I couldn't reproduce this with just one argument here, hm

**Comment by compiler-errors** [comment]
I think this is a duplicate of #108225?

**Comment by matthiaskrgr** [comment]
Ah I thought this was already fixed 😅
Probably only saw the pr...
